File: src/main/java/no/nav/bidrag/beregn/kostnadsberegnetbidrag/KostnadsberegnetBidragCore.java
package no.nav.bidrag.beregn.kostnadsberegnetbidrag;

import no.nav.bidrag.beregn.kostnadsberegnetbidrag.dto.BeregnKostnadsberegnetBidragGrunnlagCore;
import no.nav.bidrag.beregn.kostnadsberegnetbidrag.dto.BeregnetKostnadsberegnetBidragResultatCore;
import no.nav.bidrag.beregn.kostnadsberegnetbidrag.periode.KostnadsberegnetBidragPeriode;

public interface KostnadsberegnetBidragCore {

  BeregnetKostnadsberegnetBidragResultatCore beregnKostnadsberegnetBidrag(
      BeregnKostnadsberegnetBidragGrunnlagCore beregnKostnadsberegnetBidragGrunnlagCore);

  static KostnadsberegnetBidragCore getInstance() {
    return new KostnadsberegnetBidragCoreImpl(KostnadsberegnetBidragPeriode.Companion.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/kostnadsberegnetbidrag/KostnadsberegnetBidragCore.kt
package no.nav.bidrag.beregn.kostnadsberegnetbidrag

import no.nav.bidrag.beregn.kostnadsberegnetbidrag.dto.BeregnKostnadsberegnetBidragGrunnlagCore
import no.nav.bidrag.beregn.kostnadsberegnetbidrag.dto.BeregnetKostnadsberegnetBidragResultatCore
import no.nav.bidrag.beregn.kostnadsberegnetbidrag.periode.KostnadsberegnetBidragPeriode

fun interface KostnadsberegnetBidragCore {
    fun beregnKostnadsberegnetBidrag(grunnlag: BeregnKostnadsberegnetBidragGrunnlagCore): BeregnetKostnadsberegnetBidragResultatCore

    companion object {
        fun getInstance(): KostnadsberegnetBidragCore = KostnadsberegnetBidragCoreImpl(KostnadsberegnetBidragPeriode.getInstance())
    }
}
